### ISSUE TITLE
copia esto

### ISSUE THREAD

**Issue by appsdevelopmentvlneng-create** [JuanCuevasLN/E-commerce#1]
<?php

    /**
     * Responsive Child Theme functions and definitions
     * @package ResponsiveChild
     * @author Juan de la Cruz
     * @version 1.0.0
     * @since 1.0.0
     */

    // Exit if accessed directly.
    if (! defined('ABSPATH')) {
        exit;
    }

    // Cargar Funciones
    require get_stylesheet_directory() . '/inc/menu.php';
    function agregar_font_awesome()
    {
        wp_enqueue_style('font-awesome', 'https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/all.min.css');
    }
    add_action('wp_enqueue_scripts', 'agregar_font_awesome');

    require_once get_stylesheet_directory() . '/inc/replace-text-with-icons.php';

    $enqueue_dir = get_stylesheet_directory() . '/inc/enqueue/';
    foreach (glob($enqueue_dir . '*.php') as $file) {
        require_once $file;
    }

    /**
     * Cargar estilos del tema padre y del hijo
     */
    function popularfx_child_enqueue_styles()
    {
        // Carga primero el CSS del tema padre
        wp_enqueue_style(
            'popularfx-parent-style',
            get_template_directory_uri() . '/style.css'
        );

        // Luego carga el CSS del tema hijo
        wp_enqueue_style(
            'popularfx-child-style',
            get_stylesheet_directory_uri() . '/style.css',
            array('popularfx-parent-style')
        );
    }
    add_action('wp_enqueue_scripts', 'popularfx_child_enqueue_styles');

    function enqueue_filtros_ajax_script() {
        wp_enqueue_script(
            'filtros-ajax', // identificador
            get_stylesheet_directory_uri() . '/assets/js/filtros-ajax.js', // ruta
            array('jquery'), // dependencias
            '1.0.0',
            true // cargar al final del body
        );

        // Inyectamos las variables PHP para el JS
        wp_localize_script('filtros-ajax', 'wp_vars', array(
            'ajax_url' => admin_url('admin-ajax.php'),
        ));
    }
    add_action('wp_enqueue_scripts', 'enqueue_filtros_ajax_script');

    function filtrar_productos_callback(){
        $filtros = $_POST['filtros'] ?? [];

        $args = array(
            'post_type' => 'product',
            'posts_per_page' => 10,
            'post_status' => 'publish',
            'meta_query' => [],
            'tax_query' => [],
        );

        // Filtro por precio
        if (!empty($filtros['precio_min']) && !empty($filtros['precio_max'])) {
            $args['meta_query'][] = [
                'key' => '_price',
                'value' => [floatval($filtros['precio_min']), floatval($filtros['precio_max'])],
                'compare' => 'BETWEEN',
                'type' => 'NUMERIC'
            ];
        }

        // Filtro por categoría
        if (!empty($filtros['categorias'])) {
            $args['tax_query'][] = [
                'taxonomy' => 'product_cat',
                'field' => 'term_id',
                'terms' => $filtros['categorias'],
            ];
        }

        // Filtro por valoración
        if (!empty($filtros['valoracion'])) {
            $args['meta_query'][] = [
                'key' => '_wc_average_rating',
                'value' => floatval($filtros['valoracion']),
                'compare' => '>=',
                'type' => 'NUMERIC'
            ];
        }

        switch ($filtros['ordenar'] ?? 'date_desc') {
            case 'price_asc':
                $args['meta_key'] = '_price';
                $args['orderby'] = 'meta_value_num';
                $args['order'] = 'ASC';
                break;
            case 'price_desc':
                $args['meta_key'] = '_price';
                $args['orderby'] = 'meta_value_num';
                $args['order'] = 'DESC';
                break;
            case 'rating_desc':
                $args['meta_key'] = '_wc_average_rating';
                $args['orderby'] = 'meta_value_num';
                $args['order'] = 'DESC';
                break;
            case 'title_asc':
                $args['orderby'] = 'title';
                $args['order'] = 'ASC';
                break;
            default:
                $args['orderby'] = 'date';
                $args['order'] = 'DESC';
                break;
        }

        $query = new WP_Query($args);

        ob_start();

        if ($query->have_posts()) {
            echo '<div class="products-grid">';
            while ($query->have_posts()) {
                $query->the_post();
                wc_get_template_part('content', 'product');
            }
            echo '</div>';
        } else {
            echo '<p>No se encontraron productos con los filtros seleccionados.</p>';
        }

        wp_reset_postdata();
        wp_send_json_success(ob_get_clean());
    }
    add_action('wp_ajax_filtrar_productos', 'filtrar_productos_callback');
    add_action('wp_ajax_nopriv_filtrar_productos', 'filtrar_productos_callback');

jQuery(document).ready(function($) {

    function actualizarProductos() {
        console.log('Se esta organizando???');
        const loader = $('.filter-loading');
        const button = $('#id-button-filtros');
        const mainContent = $('.section-product');

        loader.show();
        button.prop('disabled', true);

        // Aquí tomas los valores de tus filtros (ajústalo a tus inputs reales)
        const filtros = {
            precio_min: $('#precio_min').val(),
            precio_max: $('#precio_max').val(),
            valoracion: $('input[name="valoracion"]:checked').val(),
            categorias: $('input[name="categoria[]"]:checked')
                .map(function() { return $(this).val(); })
                .get(),
            ordenar: $('#ordenar').val()
        };

        $.ajax({
            url: wp_vars.ajax_url,
            type: 'POST',
            data: {
                action: 'filtrar_productos',
                filtros: filtros,
            },
            success: function(response) {
                mainContent.html(response.data);

            },
            error: function(xhr, status, error) {
                console.error('Error al filtrar productos:', error);
                mainContent.html('Ocurrió un error al cargar los productos.');
            },
            complete: function() {
                loader.hide();
                button.prop('disabled', false);
            }
        })
    }

    $('#id-button-filtros').on('click', actualizarProductos);
    $('#ordenar').on('change', actualizarProductos);
});

**Comment by appsdevelopmentvlneng-create** [comment]
<?php

    /**
     * Responsive Child Theme functions and definitions
     * @package ResponsiveChild
     * @author Juan de la Cruz
     * @version 1.0.0
     * @since 1.0.0
     */

    // Exit if accessed directly.
    if (! defined('ABSPATH')) {
        exit;
    }

    // Cargar Funciones
    require get_stylesheet_directory() . '/inc/menu.php';
    function agregar_font_awesome()
    {
        wp_enqueue_style('font-awesome', 'https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/all.min.css');
    }
    add_action('wp_enqueue_scripts', 'agregar_font_awesome');

    require_once get_stylesheet_directory() . '/inc/replace-text-with-icons.php';

    $enqueue_dir = get_stylesheet_directory() . '/inc/enqueue/';
    foreach (glob($enqueue_dir . '*.php') as $file) {
        require_once $file;
    }

    /**
     * Cargar estilos del tema padre y del hijo
     */
    function popularfx_child_enqueue_styles()
    {
        // Carga primero el CSS del tema padre
        wp_enqueue_style(
            'popularfx-parent-style',
            get_template_directory_uri() . '/style.css'
        );

        // Luego carga el CSS del tema hijo
        wp_enqueue_style(
            'popularfx-child-style',
            get_stylesheet_directory_uri() . '/style.css',
            array('popularfx-parent-style')
        );
    }
    add_action('wp_enqueue_scripts', 'popularfx_child_enqueue_styles');

    function enqueue_filtros_ajax_script() {
        wp_enqueue_script(
            'filtros-ajax', // identificador
            get_stylesheet_directory_uri() . '/assets/js/filtros-ajax.js', // ruta
            array('jquery'), // dependencias
            '1.0.0',
            true // cargar al final del body
        );

        // Inyectamos las variables PHP para el JS
        wp_localize_script('filtros-ajax', 'wp_vars', array(
            'ajax_url' => admin_url('admin-ajax.php'),
        ));
    }
    add_action('wp_enqueue_scripts', 'enqueue_filtros_ajax_script');

    function filtrar_productos_callback(){
        $filtros = $_POST['filtros'] ?? [];

        $args = array(
            'post_type' => 'product',
            'posts_per_page' => 10,
            'post_status' => 'publish',
            'meta_query' => [],
            'tax_query' => [],
        );

        // Filtro por precio
        if (!empty($filtros['precio_min']) && !empty($filtros['precio_max'])) {
            $args['meta_query'][] = [
                'key' => '_price',
                'value' => [floatval($filtros['precio_min']), floatval($filtros['precio_max'])],
                'compare' => 'BETWEEN',
                'type' => 'NUMERIC'
            ];
        }

        // Filtro por categoría
        if (!empty($filtros['categorias'])) {
            $args['tax_query'][] = [
                'taxonomy' => 'product_cat',
                'field' => 'term_id',
                'terms' => $filtros['categorias'],
            ];
        }

        // Filtro por valoración
        if (!empty($filtros['valoracion'])) {
            $args['meta_query'][] = [
                'key' => '_wc_average_rating',
                'value' => floatval($filtros['valoracion']),
                'compare' => '>=',
                'type' => 'NUMERIC'
            ];
        }


        switch ($filtros['ordenar'] ?? 'date_desc') {
            case 'price_asc':
                $args['meta_key'] = '_price';
                $args['orderby'] = 'meta_value_num';
                $args['order'] = 'ASC';
                break;
            case 'price_desc':
                $args['meta_key'] = '_price';
                $args['orderby'] = 'meta_value_num';
                $args['order'] = 'DESC';
                break;
            case 'rating_desc':
                $args['meta_key'] = '_wc_average_rating';
                $args['orderby'] = 'meta_value_num';
                $args['order'] = 'DESC';
                break;
            case 'title_asc':
                $args['orderby'] = 'title';
                $args['order'] = 'ASC';
                break;
            default:
                $args['orderby'] = 'date';
                $args['order'] = 'DESC';
                break;
        }

        $query = new WP_Query($args);

        ob_start();

        if ($query->have_posts()) {
            echo '<div class="products-grid">';
            while ($query->have_posts()) {
                $query->the_post();
                wc_get_template_part('content', 'product');
            }
            echo '</div>';
        } else {
            echo '<p>No se encontraron productos con los filtros seleccionados.</p>';
        }

        wp_reset_postdata();
        wp_send_json_success(ob_get_clean());
    }
    add_action('wp_ajax_filtrar_productos', 'filtrar_productos_callback');
    add_action('wp_ajax_nopriv_filtrar_productos', 'filtrar_productos_callback');

**Comment by appsdevelopmentvlneng-create** [comment]
jQuery(document).ready(function($) {

    function actualizarProductos() {
        console.log('Se esta organizando???');
        const loader = $('.filter-loading');
        const button = $('#id-button-filtros');
        const mainContent = $('.section-product');

        loader.show();
        button.prop('disabled', true);

        // Aquí tomas los valores de tus filtros (ajústalo a tus inputs reales)
        const filtros = {
            precio_min: $('#precio_min').val(),
            precio_max: $('#precio_max').val(),
            valoracion: $('input[name="valoracion"]:checked').val(),
            categorias: $('input[name="categoria[]"]:checked')
                .map(function() { return $(this).val(); })
                .get(),
            ordenar: $('#ordenar').val()
        };

        $.ajax({
            url: wp_vars.ajax_url,
            type: 'POST',
            data: {
                action: 'filtrar_productos',
                filtros: filtros,
            },
            success: function(response) {
                mainContent.html(response.data);

            },
            error: function(xhr, status, error) {
                console.error('Error al filtrar productos:', error);
                mainContent.html('Ocurrió un error al cargar los productos.');
            },
            complete: function() {
                loader.hide();
                button.prop('disabled', false);
            }
        })
    }

    $('#id-button-filtros').on('click', actualizarProductos);
    $('#ordenar').on('change', actualizarProductos);
});

**Comment by appsdevelopmentvlneng-create** [comment]
'
            <div class="custom-cart-container">
                <a href="%s" %s data-quantity="%s" class="%s" %s> 
                    <i class="fas fa-cart-shopping"></i>
                </a>
            </div>
        ',

**Comment by JuanCuevasLN** [comment]
<?php
/**
 * The template for displaying product content in the single-product.php template
 *
 * This template can be overridden by copying it to yourtheme/woocommerce/content-single-product.php.
 *
 * HOWEVER, on occasion WooCommerce will need to update template files and you
 * (the theme developer) will need to copy the new files to your theme to
 * maintain compatibility. We try to do this as little as possible, but it does
 * happen. When this occurs the version of the template file will be bumped and
 * the readme will list any important changes.
 *
 * @see     https://woocommerce.com/document/template-structure/
 * @package WooCommerce\Templates
 * @version 3.6.0
 */

defined( 'ABSPATH' ) || exit;

global $product;

/**
 * Hook: woocommerce_before_single_product.
 *
 * @hooked woocommerce_output_all_notices - 10
 */
do_action( 'woocommerce_before_single_product' );

if ( post_password_required() ) {
	echo get_the_password_form(); // WPCS: XSS ok.
	return;
}
?>
<div id="product-<?php the_ID(); ?>" <?php wc_product_class( 'main-content', $product ); ?>>
	<div class="product-main-card">
		<div class="product-layout-grid">
			
			<div class="image-gallery-section">
				<div class="main-image-display">
					<div style="height: 500px;">
                        <?php
							/**
							* Hook: woocommerce_before_single_product_summary.
							*
							* @hooked woocommerce_show_product_sale_flash - 10
							* @hooked woocommerce_show_product_images - 20
							*/
							do_action( 'woocommerce_before_single_product_summary' );
						?>
                    </div>
				</div>	
				<div class="thumbnail-grid">
					<?php 
						do_action( 'woocommerce_product_thumbnails' );
					?>
				</div>
			</div>



		</div>
	</div>
	
	<div class="summary entry-summary">
		<?php
		/**
		 * Hook: woocommerce_single_product_summary.
		 *
		 * @hooked woocommerce_template_single_title - 5
		 * @hooked woocommerce_template_single_rating - 10
		 * @hooked woocommerce_template_single_price - 10
		 * @hooked woocommerce_template_single_excerpt - 20
		 * @hooked woocommerce_template_single_add_to_cart - 30
		 * @hooked woocommerce_template_single_meta - 40
		 * @hooked woocommerce_template_single_sharing - 50
		 * @hooked WC_Structured_Data::generate_product_data() - 60
		 */
		do_action( 'woocommerce_single_product_summary' );
		?>
	</div>

	<?php
	/**
	 * Hook: woocommerce_after_single_product_summary.
	 *
	 * @hooked woocommerce_output_product_data_tabs - 10
	 * @hooked woocommerce_upsell_display - 15
	 * @hooked woocommerce_output_related_products - 20
	 */
	do_action( 'woocommerce_after_single_product_summary' );
	?>
</div>

<?php do_action( 'woocommerce_after_single_product' ); ?>

**Comment by JuanCuevasLN** [comment]
.woocommerce main {
    border: none !important;
    margin: 0 !important;
}

body {
    font-family: 'Inter', sans-serif;
    min-height: 100vh;;
    background: linear-gradient(90deg, rgba(22, 24, 51, 1) 0%, rgba(57, 26, 96, 1) 50%, rgba(81, 26, 125, 1) 100%);
}

.container {
    max-width: 1280px;
    margin-left: auto;
    margin-right: auto;
}

/* --- MAIN CONTENT LAYOUT --- */
.main-content {
    padding-top: 2rem;
    padding-bottom: 2rem;
}

.product-main-card {
    background: rgba(255, 255, 255, 0.05); /* bg-white/5 */
    backdrop-filter: blur(20px); 
    border-radius: 1.5rem; /* rounded-2xl */
    border: 1px solid #818CF8; /* border-purple-500/20 */
    padding: 2rem;
    margin-bottom: 2rem;
}

.product-layout-grid {
    display: grid;
    grid-template-columns: 1fr;
    gap: 2rem; /* gap-12 */
}

/* --- PRODUCT IMAGE GALLERY --- */
.image-gallery-section {
    display: flex;
    flex-direction: column;
    gap: 1.5rem;
}

.main-image-display {
    background:  rgba(255, 255, 255, 0.1); /* bg-white/10 */
    border-radius: 0.75rem; /* rounded-xl */
    padding: 0.75rem;
    border: 1px solid #6366F1; /* border-purple-400/20 */
}

.main-image-display img {
    width: 100%;
    height: 100%;
    object-fit: cover;
}

**Comment by JuanCuevasLN** [comment]
:root {
            /* Colores */
            --color-dark-navy: #161733; /* Black/20 */
            --color-dark-purple-mid: #361d4b; /* Purple 500/20 for background */
            --color-dark-purple-dark: #3c1e63; /* Purple 500 for accents */
            --color-purple-200: #C4B5FD;
            --color-purple-300: #A5B4FC;
            --color-purple-400: #818CF8;
            --color-purple-600: #6366F1; /* Purple 600 for gradient/buttons */
            --color-pink-600: #EC4899; /* Pink 600 for gradient */
            --color-yellow-400: #FACC15;
            --color-green-400: #4ADE80;
            --color-white-5: rgba(255, 255, 255, 0.05); /* White/5 */
            --color-white-10: rgba(255, 255, 255, 0.1); /* White/10 */
            --color-white-20: rgba(255, 255, 255, 0.2); /* White/20 */
            
            /* Espaciado */
            --space-1: 0.25rem;
            --space-2: 0.5rem;
            --space-3: 0.75rem;
            --space-4: 1rem;
            --space-6: 1.5rem;
            --space-8: 2rem;
        }
